public/js/register.js

```
@@ -0,0 +1,129 @@
+const registerForm = document.getElementById("registerForm");
+
+if (registerForm) {
+  const firstName = document.getElementById("firstName");
+  const lastName = document.getElementById("lastName");
+  const email = document.getElementById("emailRegister");
+  const firstPassword = document.getElementById("firstPassword");
+  const secondPassword = document.getElementById("confirmPassword");
+  const age = document.getElementById("age");
+  const gender = document.getElementById("gender");
+  const role = document.getElementById("role");
+  const error = document.getElementById("registerError");
+
+  registerForm.addEventListener("submit", (event) => {
+    let firstNameValue = firstName.value;
+    let lastNameValue = lastName.value;
+    let emailValue = email.value;
+    let firstPasswordValue = firstPassword.value;
+    let secondPasswordValue = secondPassword.value;
+    let ageValue = age.value;
+    let genderValue = gender.value;
+    let roleValue = role.value;
+
```

In all client validation, upon resubmit i think you should hide the error message. This is optional but will give a better understanding to the user.